media-libs/openimageio/openimageio-3.0.8.1.ebuild

```diff
@@ -0,0 +1,444 @@
+
+RESTRICT="!test? ( test )"
+
+BDEPEND="
+	jpeg2k? ( app-arch/unzip )
+	doc? (
+		app-text/doxygen
+		dev-texlive/texlive-bibtexextra
+		dev-texlive/texlive-fontsextra
+		dev-texlive/texlive-fontutils
+		dev-texlive/texlive-latex
+		dev-texlive/texlive-latexextra
+	)
+"
+
+# >=media-libs/libuhdr-1.3
+RDEPEND="
+	app-arch/bzip2:=
+	dev-cpp/robin-map
+	dev-libs/pugixml
+	media-libs/libheif:=
+	media-libs/libjpeg-turbo:=
+	media-libs/libpng:=
+	media-libs/libwebp:=
+	media-libs/opencolorio:=
+	media-libs/openexr:=
+	media-libs/tiff:=
+	sys-libs/zlib:=
+	dicom? ( sci-libs/dcmtk )
+	ffmpeg? ( media-video/ffmpeg:= )
+	fits? ( sci-libs/cfitsio:= )
+	gif? ( media-libs/giflib:= )
+	jpeg2k? ( media-libs/openjpeg:= )
+	jpegxl? ( media-libs/libjxl:= )
+	opencv? ( media-libs/opencv:= )
+	openvdb? (
+		dev-cpp/tbb:=
+		media-gfx/openvdb:=
+	)
+	ptex? ( media-libs/ptex:= )
+	python? (
+		${PYTHON_DEPS}
+		$(python_gen_cond_dep '
+			dev-python/numpy:=[${PYTHON_USEDEP}]
+			dev-python/pybind11[${PYTHON_USEDEP}]
+		')
+	)
+	gui? (
+		media-libs/libglvnd
+		dev-qt/qtbase:6[gui,widgets,opengl]
+	)
+	raw? ( media-libs/libraw:= )
+	truetype? ( media-libs/freetype )
+"
+DEPEND="
+	dev-libs/imath:=
+	dev-libs/libfmt:=
+	${RDEPEND}
+	test? ( media-fonts/droid )
+"
+
+DOCS=(
+	CHANGES.md
+	CREDITS.md
+	README.md
+)
+
+PATCHES=(
+	# "${FILESDIR}/${PN}-2.5.8.0-fix-tests.patch"
+	"${FILESDIR}/${PN}-2.5.12.0-heif-find-fix.patch"
+	"${FILESDIR}/${PN}-2.5.18.0-tests-optional.patch"
+	# in src_prepare
+	# "${FILESDIR}/${PN}-2.5.12.0_heif_test.patch"
+)
+
+pkg_setup() {
+	use python && python-single-r1_pkg_setup
+}
+
+src_prepare() {
+	if ! use dicom; then
+		rm "src/dicom.imageio" -r || die
+	fi
+
+	if ! use gif; then
+		rm src/gif.imageio -r || die
+	fi
+
+	if ! use jpeg2k; then
+		rm src/jpeg2000.imageio -r || die
+	fi
+
+	if ! use raw; then
+		rm src/raw.imageio -r || die
+	fi
+
+	cmake_src_prepare
+	cmake_comment_add_subdirectory src/fonts
+
+	if use test ; then
+		ln -s "${WORKDIR}/OpenImageIO-images-${TEST_OIIO_IMAGE_COMMIT}" "${WORKDIR}/oiio-images" || die
+		ln -s "${WORKDIR}/openexr-images-${TEST_OEXR_IMAGE_COMMIT}" "${WORKDIR}/openexr-images" || die
+
+		if use fits; then
+			mkdir -p "${WORKDIR}/fits-images/"{ftt4b,pg93} || die
+			for a in ${A}; do
+				if [[ "${a}" == file*.fits ]]; then
+					ln -s "${DISTDIR}/${a}" "${WORKDIR}/fits-images/ftt4b/" || die
+				fi
+				if [[ "${a}" == tst*.fits ]]; then
+					ln -s "${DISTDIR}/${a}" "${WORKDIR}/fits-images/pg93/" || die
+				fi
+			done
+		fi
+
+		if use jpeg2k; then
+			ln -s "${WORKDIR}/J2KP4files" "${WORKDIR}/j2kp4files_v1_5" || die
+		fi
+
+		cp testsuite/heif/ref/out-libheif1.1{2,5}-orient.txt || die
+		eapply "${FILESDIR}/${PN}-2.5.12.0_heif_test.patch"
+	fi
+
+	mkdir "${T}/cmake" || die
+
+	local libdir="${ESYSROOT}/usr/$(get_libdir)"
+
+	# generate our own WebPConfig.cmake via pkg-config # 937031
+	cat <<-EOF > "${T}/cmake/WebPConfig.cmake" || die
+	set(WebP_VERSION $(pkg-config --modversion libwebp))
```

Comment:
This is quite lengthy and also probably wrong as it doesn't respect the tc-get* handling for pkg-config.

Wouldn't it be simpler anyways to just have cmake itself use pkg_check_modules and then creat aliased targets for the names that the upstream code looks for?

Reply:
I really, really don't want to ship a large bespoke .cmake file, sorry. It's the kind of thing where it will absolutely go wrong, and it'll drive someone crazy debugging it.

Reply:
Please do yourself some favours by not responding with "No" then.

Reply:
It's part of the ebuild...